ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/EstablishedPeers.hs

```diff
@@ -410,6 +528,76 @@ aboveTarget actions
   = GuardedSkip Nothing
 
 
+aboveTargetBigLedgerPeers :: forall peeraddr peerconn m.
+                             (MonadSTM m, Ord peeraddr)
+                          => PeerSelectionActions peeraddr peerconn m
+                          -> MkGuardedDecision peeraddr peerconn m
+aboveTargetBigLedgerPeers actions
+                          policy@PeerSelectionPolicy {
+                            policyPickWarmPeersToDemote
+                          }
+                          st@PeerSelectionState {
+                            bigLedgerPeers,
+                            establishedPeers,
+                            activePeers,
+                            inProgressDemoteWarm,
+                            inProgressPromoteWarm,
+                            targets = PeerSelectionTargets {
+                                        targetNumberOfEstablishedBigLedgerPeers
+                                      }
+                          }
+    -- Are we above the target for number of established peers?
+    -- Or more precisely, how many established peers could we demote?
+    -- We only want to pick established peers that are not active, since for
+    -- active one we need to demote them first.
+  | let numEstablishedBigLedgerPeers, numBigLedgerPeersToDemote :: Int
+        numEstablishedBigLedgerPeers = Set.size $ EstablishedPeers.toSet establishedPeers
+                               `Set.intersection` bigLedgerPeers
+
+        numActiveBigLedgerPeers      = Set.size $ activePeers
+                               `Set.intersection` bigLedgerPeers
+
+        numBigLedgerPeersToDemote    = min ( numEstablishedBigLedgerPeers
+                                           - targetNumberOfEstablishedBigLedgerPeers)
+                                           ( numEstablishedBigLedgerPeers
+                                           - numActiveBigLedgerPeers)
+                                     - Set.size inProgressDemoteWarm
+                                     - Set.size inProgressPromoteWarm
```

Comment:
Add a comment here: we want to demote peers towards the target but we don't want to pick active peers.